test_runner/src/main/kotlin/ftl/json/MatrixMap.kt

```diff
@@ -9,12 +9,12 @@ import ftl.util.MatrixCanceledError
 import ftl.util.MatrixState
 
 class MatrixMap(
-    val map: Map<String, SavedMatrix>,
+    val map: MutableMap<String, SavedMatrix>,
```

Comment:
Interesting, i agree that MatrixMap is a good candidate for the data class conversion.
The question is should i do it apart of this ticket as it is heavily used within the class or should it be apart of another ticket. 
As this ticket now has two dependancies, i feel that it might be worth to make another ticket and work on it after this ticket is merged as it will require some updates across the code base.

Reply:
I don't like MutableMaps too, but changes in this PR are a good tradeoff for it.